Deprecate timeout.tcp_read AD/LDAP realm setting

albertzaharovits opened the pull request:
The `timeout.tcp_read` AD/LDAP realm setting, despite the low-level allusion, controls the time interval the realms wait for a response for a query (search or bind). If the connection to the server is synchronous (un-pooled) the response timeout is analogous to the tcp read timeout. But the tcp read timeout is irrelevant in the common case of a pooled connection (when a Bind DN is specified).

The `timeout.tcp_read` qualifier is hereby deprecated in favor of `timeout.response`.

In addition, the default value for both `timeout.tcp_read` and `timeout.response` is that of `timeout.ldap_search`, instead of the `5s` (but the default for `timeout.ldap_search` is still `5s`). The `timeout.ldap_search` defines the server-controlled timeout of a search request. There is no practical use case to have a smaller `tcp_read` timeout compared to `ldap_search` (in this case the request would time-out on the client but continue to be processed on the server). The proposed change aims to simplify configuration so that the more common configuration change, adjusting `timeout.ldap_search` up, has the expected result (no timeout during searches) without any additional modifications.

Closes https://github.com/elastic/elasticsearch/issues/46028

albertzaharovits committed:
Done

elasticmachine commented:
Pinging @elastic/es-security

jkakavas commented on docs/reference/settings/security-settings.asciidoc:
```diff
@@ -693,9 +700,16 @@ An `s` at the end indicates seconds, or `ms` indicates milliseconds.
-An `s` at the end indicates seconds, or `ms` indicates milliseconds.
-Defaults to `5s` (5 seconds ).
+deprecated[7.5] The TCP read timeout period after establishing an LDAP
+connection.  This is equivalent and is deprecated in favor of
```
```suggestion
connection.  This is equivalent to and is deprecated in favor of
```

jkakavas commented on docs/reference/settings/security-settings.asciidoc:
```diff
@@ -445,9 +445,16 @@ An `s` at the end indicates seconds, or `ms` indicates milliseconds.
-An `s` at the end indicates seconds, or `ms` indicates milliseconds.
-Defaults to `5s` (5 seconds ).
+deprecated[7.5] The TCP read timeout period after establishing an LDAP
+connection.  This is equivalent and is deprecated in favor of
```
```suggestion
connection.  This is equivalent to and is deprecated in favor of
```

jkakavas commented on docs/reference/settings/security-settings.asciidoc:
```diff
@@ -445,9 +445,16 @@ An `s` at the end indicates seconds, or `ms` indicates milliseconds.
+`timeout.ldap_search`.
 
 `timeout.ldap_search`::
 The LDAP Server enforced timeout period for an LDAP search.
```
Is it worth adding one more sentence here or rewriting this somehow to explain that this is sent  the server as part of the request ? The initial question I can think of when someone reads this is "If this a server setting, why do I set it in the client" ? WDYT ?

albertzaharovits replied:
Refactored to:
```
`timeout.ldap_search`::
The timeout period for an LDAP search. The value is specified in the request
and is enforced by the receiving LDAP Server.
An `s` at the end indicates seconds, or `ms` indicates milliseconds.
Defaults to `5s` (5 seconds ).
```

jkakavas reviewed (approved):
LGTM

albertzaharovits committed:
Update docs/reference/settings/security-settings.asciidoc

Co-Authored-By: Ioannis Kakavas <ikakavas@protonmail.com>

albertzaharovits committed:
Update docs/reference/settings/security-settings.asciidoc

Co-Authored-By: Ioannis Kakavas <ikakavas@protonmail.com>

albertzaharovits commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into unbound_id_response_timeout

albertzaharovits committed:
refactored ldap_search explanation

albertzaharovits committed:
Merge branch 'master' into unbound_id_response_timeout

albertzaharovits commented:
Note, I plan to follow-up with the deprecation check and the breaking change PRs when this is merged and backported.

tvernum commented on docs/reference/migration/migrate_8_0/security.asciidoc:
```diff
@@ -41,3 +41,12 @@ realm directly.
+
+[float]
+[[ldap-ad-realms-tcp-read-timeout-removed]]
+==== The `timeout.tcp_read` AD and LDAP realm settings have been removed
```
I'm confused by the `removed` here. This PR doesn't remove them - I think this whole docs piece should be held over to the removal PR.

albertzaharovits replied:
Agreed. I'll hold this off for the actual removal PR.

tvernum commented on docs/reference/settings/security-settings.asciidoc:
```diff
@@ -445,12 +445,20 @@ An `s` at the end indicates seconds, or `ms` indicates milliseconds.
+connection. This is equivalent to and is deprecated in favor of
+`timeout.response` and they cannot be used simultaneously. An `s` at the end
+indicates seconds, or `ms` indicates milliseconds. Defaults to the value of
+`timeout.ldap_search`.
```
I'd propose that we remove the "Defaults to ..." from here, since it is the deprecated setting, it doesn't really have a default.

albertzaharovits replied:
Good point! If you one doesn't specify the deprecated settings, it's the new settings that takes a default value, the old one is just deprecated.

albertzaharovits committed:
Merge branch 'master' into unbound_id_response_timeout

albertzaharovits committed:
Tim's review!

albertzaharovits commented:
@tvernum I have acted on your objections, please take another look.

albertzaharovits commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into unbound_id_response_timeout

droberts195 committed:
[ML] Use CSV ingest processor in find_file_structure ingest pipeline (#51492)

Changes the find_file_structure response to include a CSV
ingest processor in the ingest pipeline it suggests.

Previously the Kibana file upload functionality parsed CSV
in the browser, but by parsing CSV in the ingest pipeline
it makes the Kibana file upload functionality more easily
interchangable with Filebeat such that the configurations
it creates can more easily be used to import data with the
same structure repeatedly in production.

dnhatn committed:
Add test verify replica allocator with sync_id (#51512)

We no longer issue new sync_ids in 8.0, but we still need to make sure 
that the replica allocator prefers copies with matching sync_id. This
commit adds tests for that.

Relates #50776

pugnascotia committed:
Formatting: keep simple if / else on the same line (#51526)

Previous the formatter was breaking simple if/else statements (i.e.
without braces) onto separate lines, which could be fragile because the
formatter cannot also introduce braces. Instead, keep such expressions
on the same line.

albertzaharovits committed:
Nits

albertzaharovits committed:
Merge branch 'master' into unbound_id_response_timeout

albertzaharovits commented:
Aaaaargh, apologies for the bogus commit message!
The merge failed, and I compulsively clicked "Try again" although I knew it's wrong 😞
